pkg/resource/key/hooks.go

```diff
@@ -0,0 +1,43 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//     http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package key
+
+import (
+	"strconv"
+
+	svcapitypes "github.com/aws-controllers-k8s/kms-controller/apis/v1alpha1"
+)
+
+const (
+	DefaultDeletePendingWindowInDays = 7
+)
+
+// getDeletePendingWindowInDays returns the pending window (in days) as
+// determined by the annotation on the object, or the default value otherwise.
+func (rm *resourceManager) getDeletePendingWindowInDays(
```

Comment:
nit: This is very simple but unit test can be helpful. (also just a good habit)
nit: this need not be a method on `*resourceManager`. This can be a function by itself.
nit: rather than `*resource`, you can take into input `metav1.Object` because that's all this function cares about.